src/Interpreters/InterpreterSystemQuery.cpp

```diff
@@ -1161,6 +1166,81 @@ void InterpreterSystemQuery::waitLoadingParts()
     }
 }
 
+
+void InterpreterSystemQuery::loadPrimaryKeys()
+{
+    if (!table_id.empty())
+    {
+        getContext()->checkAccess(AccessType::SYSTEM_LOAD_PRIMARY_KEY, table_id.database_name, table_id.table_name);
+        StoragePtr table = DatabaseCatalog::instance().getTable(table_id, getContext());
+
+        if (auto * merge_tree = dynamic_cast<MergeTreeData *>(table.get()))
+        {
+            LOG_TRACE(log, "Loading primary keys for table {}", table_id.getFullTableName());
+            try
+            {
+                merge_tree->loadPrimaryKeys();
+            }
+            catch (const Exception & ex)
+            {
+                LOG_ERROR(log, "Failed to load primary keys for table {}: {}", table_id.getFullTableName(), ex.message());
+            }
+        }
+        else
+        {
+            throw Exception(
+                ErrorCodes::BAD_ARGUMENTS, "Command LOAD PRIMARY KEY is supported only for MergeTree tables, but got: {}", table->getName());
+        }
+    }
+    else
+    {
+        getContext()->checkAccess(AccessType::SYSTEM_LOAD_PRIMARY_KEY);
+
+        // Define a thread pool for parallel processing
+        auto & thread_pool = DB::getActivePartsLoadingThreadPool().get();
+
+        // Cap the number of concurrent tasks based on thread pool availability
+        size_t max_tasks = std::min(thread_pool.maxConcurrency(), DatabaseCatalog::instance().getDatabases().size());
```

Comment:
Did you mean to call `getMaxThreads()` instead of `maxConcurrency()`? Or where does the latter come from?
Code doesn't even compile with `maxConcurrency()` :thinking: